src/enumerable/enumerable_gather.jl

```diff
@@ -0,0 +1,80 @@
+struct EnumerableGather{T,S,F,I,A} <: Enumerable
+    source::S
+    fields::F
+    indexFields::I
+    savedFields::A
+    key::Symbol
+    value::Symbol
+end
+
+struct Not{T}
+    val::T
+end
+
+function gather(source::Enumerable, args...; key::Symbol = :key, value::Symbol = :value)
+    fields = fieldnames(eltype(source))
+    if length(args) > 0
+        indexFields = ()
```

Comment:
I think in general it would be more efficient to use a `Vector` here, and then convert to the `Vector` to a tuple once the final list exists. But it probably doesn't matter much, so we can leave as is for now and maybe fix in a follow up PR.